config/environments/production.rb

```diff
@@ -69,7 +69,7 @@
   config.log_tags = [:request_id]
 
   # Use a different cache store in production.
-  config.cache_store = :mem_cache_store
+  config.cache_store = :mem_cache_store, nil, { namespace: ENV.fetch("MEMCACHE_KEY_PREFIX", "whitehall"), compress: true } unless ENV["HEROKU_APP_NAME"]
```

Comment:
Would you be to explain why we need namespacing as I'm not too clear? 

Also we won't need a heroku conditional for this app.

Reply:
Ah that's fab, I'll get rid of the Heroku bit, thanks!

The prefix is needed so that we don't have to worry about name clashes with other apps using the same memcache.

(We could have separate memcaches per app, but we don't want that because what we'd theoretically gain in resource isolation doesn't outweigh the admin overhead and complexity and waste that it'd entail, plus we know the app is well-behaved in the face of failing memcache requests anyway so even the resource isolation thing is moot.)

Reply:
Done.

Reply:
Ah ok - yeah separate memcaches per app seem more consistent with our direction following https://github.com/alphagov/govuk-rfcs/blob/main/rfc-143-split-database-instances.md. I know we've just got a looming challenge too to separate out the namespaced redis instances since support ends for that with Sidekiq 7. Do agree that memcache is more friendly on failures. Is there a place that documents the shared memcache decision just so it's clear why it goes a different route to our other shared data direction of travel? Not really a concern for this PR.

Any chance you could update the commit message to state that the purpose of the namespace is so that it can be using a shared instance? that's the part I needed explaining.

Reply:
This is an improvement that I've been working on in my spare time — I'm not really able to handle the bureaucracy, sorry.

Reply:
Apologies for being a bit grumpy there earlier. I've written an extensive description that explains the rationale, the wider context into which this change fits and the direction of travel.

That's interesting that Sidekiq 7 is dropping support for the redis-namespace wrapper. (Looking at the implementation of redis-namespace I can see why, but there's clearly a need for it to be a first class feature of Sidekiq. Redis's numbered "databases" don't solve the same problem at all because there's no sensible way to map the numerical IDs automatically.) Sounds like an ideal opportunity to switch to in-cluster Redises per app: less toil, more developer-friendly, less cost, and we'd be able to test with the same software that we use in prod. The argument for Elasticache for Redis was that running our own was more hassle so it was worth the expense and the config complexity, but the reverse is true now.